cmd/cli/serve/serve.go

```diff
@@ -50,6 +50,13 @@ var (
 `))
 )
 
+const (
+	NameFlagName        = "name"
+	NameFlagDescription = `The node's name.
+If unset, it will be read from .bacalhau/system_metadata.yaml, or automatically generated if no name exists.
+If set, the provided value will override the current name in .bacalhau/system_metadata.yaml.`
```

Comment:
it shouldn't. We want the same as the existing behaviour. A node `--name` is only applied the first time you instantiate a node, will be persisted afterwards, and then the persisted node name will be used and the flag will be ignored

In the future we might fail if the provided name doesn't match the one in system_metadata, but we are not sure if that is the right behaviour and no need for us to change the existing behaviour at this time

In other words, we only use this flag before calling `config.GenerateNodeID()`

Reply:
If the user provides the name flag and a value is already present in `system_metadata.yaml` should we fail, ignore the flag, or ignore the flag and warn?

Reply:
I've addressed this and added a warning if the flag is used and a value is already present